src/urbit/app/chess.hoon

```diff
@@ -627,7 +632,7 @@
       =/  fen  (position-to-fen position.game-state)
       =/  cards  ^-  (list card)
         :~  :*  %give  %fact  ~[/game/(scot %da u.game-id)/updates]
-                %chess-update  !>([%position u.game-id fen])
+                %chess-update  !>([%position u.game-id fen ''])
```

Comment:
The `''` here is where the `san` position update would go if there was one. Do I need to include this here? It would be nice if I didn't, as `useChessStore` currently needs some conditional logic to ignore a `''` position, so as not to start each game's `displayMoves` array with an empty index.

Reply:
It might be more correct to change `''` to a `~` and check how `enjs:format` handles that, then update `useChessStore` accordingly.

Reply:
Hmmm - if it's this annoying to deal with appending/removing moves in the frontend, maybe each position update should call `algebraicize-and-number` and return the entire list of moves, which can be stored in `ActiveGameInfo`? It's not the most efficient way to do this, but it might be significantly easier to code. I know I recommended against it earlier, but I didn't realize at the time how annoying it would be to manage the moves for each game in the frontend.

About your earlier comment, "Sending a whole new list of moves in the '%position' update would require scrying '[%x %game @ta ~]' from 'update.hoon'": no, you could call `(algebraicize-and-number game.game-state)` in `try-move` and `%chess-debug-inject`, and use the empty list (`~`) in `[%game @ta %updates ~]`.

Reply:
Hmmm - but then after #47 goes in, that would need to be extended to sending the entire list of moves and their FENs after each new move, which is becoming really awful...

Reply:
So I guess the frontend will need to a new type of object, which contains not just the `ActiveGameInfo`, but also an `Array` of move data, which is a pair of move SAN and FEN after that move. Getting a position update involves updating the `ActiveGameInfo` and adding another move to this `Array`. Accepting/receiving an undo involves removing a move from this `Array`.

Reply:
The problem with `algebraicize-and-number` is that it sends each cord like `'1. f3'` or `'1. f3 e6'`, so to make each ply clickable I'd need to parse and split each string as it comes in to separate them into clickable `span`s.

Reply:
The specific concern I had was that `''` rather than `~` feels like a weird kludge. `~`s are returned throughout the app and that looks/feels fine. I'll amend it now and see how the frontend reacts.

Reply:
Yeah, ignore `algebraicize-and-number` -just send one SAN move at a time. To make this work for subscriptions from the frontend to in-progress games is that they'll need to send a stream of update cards for each move that has occurred so far, each containing the move SAN and FEN.

Reply:
That sounds good. The `''` to `~` change was fine so I'll revisit this issue once the undo functionality is merged, putting the SAN and FENs together in the updates.